d2graph/d2graph.go

```diff
@@ -732,6 +742,10 @@ func (obj *Object) Connect(srcID, dstID []string, srcArrow, dstArrow bool, label
 	src := srcObj.EnsureChild(srcID)
 	dst := dstObj.EnsureChild(dstID)
 
+	if src.outerSequenceDiagram() != dst.outerSequenceDiagram() {
+		return nil, errors.New("connections within sequence diagrams can connect only to other objects within the same sequence diagram")
```

Comment:
it could also be that objects inside a sequence diagram can't receive connections from the outside

Reply:
this error message would cover that. there is no such thing as "receive connection", as that implies directionality. an edge connects two objects.